### PR TITLE
test(sample-issue-tracker): business-logic bun coverage (#26)

### LINKED ISSUE #26: SampleIssueTracker: end-to-end Bun tests for generated business logic
**Current state:** \`sample-issue-tracker\` has 51 passing Bun tests ([as of 2026-04-11](https://github.com/danfma/metano/pull/13)), but the coverage is focused on **generated shape correctness** (can the TS compile, do imports resolve, do classes instantiate). What's missing is **business logic coverage** of the generated code — asserting that the transpiled workflow actually behaves the way the C# source intends.

## Goal

Add end-to-end Bun test files that exercise the business rules of the transpiled \`SampleIssueTracker\` types, not just their compilation:

- [ ] \`IssueWorkflow\`: state transitions (Backlog → InProgress → Done), rejected transitions, invariant checks
- [ ] \`Issue\`: comment addition preserves immutability (records), priority changes compute correctly
- [ ] \`IssueService\` / \`InMemoryIssueRepository\`: CRUD round-trips, filter/sort queries return expected shapes
- [ ] \`IssueQueries\`: LINQ queries (\`openIssues\`, grouping by priority, etc.) produce the same results as their C# counterparts
- [ ] \`Sprint\` / \`Planning\`: capacity rules, sprint-to-issue linking
- [ ] \`OperationResult\` success/failure paths propagate correctly across service calls
- [ ] Branded IDs (\`UserId\`, \`IssueId\`): value-level equality, construction via \`.newUuid()\`, serialization round-trip

## Why

The current tests catch \"the transpiler regressed how it emits X\". These tests would catch \"the semantics of the generated X diverged from the C# source\" — a harder class of bug that won't show up until a consumer actually runs the logic.

## Out of scope

Rewriting or expanding the C# \`SampleIssueTracker\` source itself. This is purely a test-writing task against the **existing** generated code.

Part of #14.

### PR DESCRIPTION
**Pull request by danfma**
## Summary

Pure test-writing PR — zero transpiler code changes. Adds 5 new bun test files exercising **semantic** parity between generated TS and the C# source for SampleIssueTracker:

- \`issues/domain/issue-workflow-matrix.test.ts\` — exhaustive \`from × to\` transition matrix
- \`issues/domain/issue-mutations.test.ts\` — UpdatedAt / CreatedAt / Id / Type invariants, comment order, transition comment shape
- \`issues/application/issue-service-workflows.test.ts\` — chained ops + OperationResult not-found propagation across \`assign\` / \`planSprint\` / \`transition\` / \`addComment\`
- \`issues/application/issue-queries-edges.test.ts\` — empty list, closed-variant exclusion, ordering, limit edge cases
- \`planning/domain/sprint.test.ts\` — zero tests before; covers durationDays, plan/unplan, rename/reschedule

## New issues surfaced

- **#90** — C# DateOnly comparisons lower to raw JS \`>=\` / \`<=\` but Temporal.PlainDate rejects those operators at runtime. \`Sprint.isActiveOn\` trips it. One test skipped with inline reference; unskip after lowering lands.

## Test plan

- [x] 633/633 TUnit green (unchanged — no transpiler code touched)
- [x] 141/142 bun tests pass in sample-issue-tracker (+72 new; 1 skip tracked in #90)
- [x] sample-todo + sample-todo-service bun suites unchanged

## Net delta

5 new files, +562 / -0.

Closes #26

### CODE REVIEW

File: targets/js/sample-issue-tracker/test/issues/domain/issue-mutations.test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+describe("Issue mutations", () => {
+  test("CreatedAt and UpdatedAt start within the same second", () => {
+    const issue = makeIssue();
+    // Each field initializes via its own `Temporal.Now.zonedDateTimeISO()`
+    // call, which produces distinct nanosecond values even when evaluated
+    // in the same synchronous pass. Assert they land in the same second
+    // instead of pinning exact equality.
+    const createdSec = issue.createdAt.epochNanoseconds / 1_000_000_000n;
+    const updatedSec = issue.updatedAt.epochNanoseconds / 1_000_000_000n;
+    expect(createdSec).toBe(updatedSec);
```
[the remark this answers is not in the document]
Comment (danfma) [reply targets/js/sample-issue-tracker/test/issues/domain/issue-mutations.test.ts]: Fixed in eb68e29. Applied the suggested absolute-delta form (<= 1 second). Same-whole-second bucket would flake once per second when the boundary falls between the two Temporal.Now calls.

File: targets/js/sample-issue-tracker/test/issues/domain/issue-mutations.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  });
+
+  test("rename advances UpdatedAt", () => {
+    const issue = makeIssue();
+    const before = issue.updatedAt.epochNanoseconds;
+    issue.rename("New title");
+    expect(issue.updatedAt.epochNanoseconds >= before).toBe(true);
+    // Plus the rename itself took effect.
+    expect(issue.title).toBe("New title");
```
[the remark this answers is not in the document]
Comment (danfma) [reply targets/js/sample-issue-tracker/test/issues/domain/issue-mutations.test.ts]: Fixed in eb68e29. Renamed test to 'does not decrease UpdatedAt' — matches the actual '>=' assertion. Temporal.Now can return the same nanosecond for two synchronous calls so strict '>' would flake.

File: targets/js/sample-issue-tracker/test/issues/application/issue-service-workflows.test.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    const created = await service.createAsync(
+      "Task",
+      "x",
+      IssueType.Story,
+      IssuePriority.Medium,
+    );
+    // Backlog → Done is disallowed; the underlying Issue.transitionTo
+    // throws and the async wrapper propagates. OperationResult shape
+    // doesn't swallow runtime exceptions today — this test pins that
+    // expectation so a future change to the dispatcher surfaces.
```
[the remark this answers is not in the document]
Comment (danfma) [reply targets/js/sample-issue-tracker/test/issues/application/issue-service-workflows.test.ts]: Fixed in eb68e29. Retitled 'invalid status transition surfaces as a thrown Error' (plain JS Error since C# InvalidOperationException lowers to Error). Added regex assertion against the message (/Cannot transition issue from/) to pin the contract.
